Source/Core/Common/File.h

```diff
@@ -13,10 +13,43 @@
 
 namespace File
 {
+class ReadOnlyFile
+{
+public:
+  template <typename T>
+  bool ReadArray(T* data, size_t length, size_t* pReadElements = nullptr)
+  {
+    return Read(data, sizeof(T), length, pReadElements)));
```

Comment:
Yeah, I got some help with it on IRC. The non-Windows error messages helped too.

I've fixed this locally, but there are still problems, so I'll wait with pushing more until I have something that actually works.